src/utils.rs

```diff
@@ -68,25 +58,20 @@ pub async fn create_embed_response(
     interaction: &mut ApplicationCommandInteraction,
     embed: CreateEmbed,
 ) -> Result<(), ParrotError> {
-    interaction
+    let response: Result<(), ParrotError> = interaction
         .create_interaction_response(&http, |response| {
             response
                 .kind(InteractionResponseType::ChannelMessageWithSource)
-                .interaction_response_data(|message| message.add_embed(embed))
+                .interaction_response_data(|message| message.add_embed(embed.clone()))
         })
         .await
-        .map_err(Into::into)
-}
-
-pub async fn create_embed_followup(
-    http: &Arc<Http>,
-    interaction: &mut ApplicationCommandInteraction,
-    embed: CreateEmbed,
-) -> Result<Message, ParrotError> {
-    interaction
-        .create_followup_message(&http, |followup| followup.add_embed(embed))
-        .await
-        .map_err(Into::into)
+        .map_err(Into::into);
+    match response {
+        Ok(val) => Ok(val),
+        Err(..) => edit_embed_response(http, interaction, embed)
```

Comment:
Can we use some more concrete error

Reply:
I wanted to use Serenity Error::Http but since it gets remapped to a ParrotError I couldn't figure out a good way to handle the match properly.

Reply:
You can pattern match it I think, unwrapping the http serenity error from here: https://github.com/aquelemiguel/parrot/blob/main/src/errors.rs#L25

Reply:
@joao-conde How would you go about making sure this specific Error::Http is related to an interaction response already existing? Maybe through [serenity::http::error::DiscordJsonError](https://docs.rs/serenity/latest/serenity/http/error/struct.DiscordJsonError.html) code property? This might be overkill.

Reply:
I dont know, @StaticRocket said he wanted to use Serenity Error HTTP, I am just saying he can since it is wrapped inside the ParrotError. 

This heavily depends on the error returned. @StaticRocket in the original issue, could you provide an example of the error you returned?

Reply:
@joao-conde an example was provided in the original ticket here: https://github.com/aquelemiguel/parrot/issues/201
Turns out it is a DiscordJsonError.

Reply:
The real headache is that this gets remapped into ParrotError::Serenity(serenity::http::error::DiscordJsonError{code: 40060, message: "Interaction has already been acknowledged.", errors: []} and I have no clue how to match that. Going to try working around the map_err and wrapping it later.

Reply:
Scratch that, figured it out.